pkg/chains/objects/objects.go

```diff
@@ -40,14 +43,158 @@ type Object interface {
 	runtime.Object
 }
 
-// Result is a generic key value store containing the results
+// GenericResult is a generic key value store containing the results
 // of Tekton operations. (eg. PipelineRun and TaskRun results)
-type Result struct {
+type GenericResult interface {
+	// GetName returns the name associated with the result.
+	GetName() string
+
+	// GetStringValue returns the string value of the result.
+	GetStringValue() string
+
+	// GetObjectValue returns the object value for the specified field.
+	GetObjectValue(field string) string
+
+	// ObjectValueIsNil checks if the object value is nil.
+	ObjectValueIsNil() bool
+}
+
+type GenericProvenance interface {
+	IsNil() bool
+	RefSourceIsNil() bool
+
+	GetRefSourceURI() string
+	GetRefSourceDigest() common.DigestSet
+	GetRefSourceEntrypoint() string
+
+	FeatureFlagsIsNil() bool
+	GetFeatureFlags() *config.FeatureFlags
```

Comment:
It might make more sense to define these on the upstream types. 🤷 

Otherwise since these are roughly equivalent it's probably fine to pick one to use everywhere.

Reply:
I have removed the GenericProvenance and associated interface methods, instead opting to use *v1.Provenance in all methods and converting the v1beta1.Provenance -> v1.Provenance for usage in TektonObject(s).  Doing so brings up a potential issue you identified here https://github.com/tektoncd/chains/pull/1016#discussion_r1431754916 related to `ConfigSource` not being present in v1.Provenance but I'm a bit confused what if any the downsides are as v1beta1.Provenance.ConfigSource doesn't actually seemed to be used in any code @ HEAD.

Reply:
I have added support for ConfigSource during the v1beta1 -> v1 Provenance conversion process (NOTE: this doesn't seem to be present @ HEAD)